celery/backends/consul.py

```diff
@@ -40,15 +39,33 @@ def __init__(self, *args, **kwargs):
 
         if self.consul is None:
             raise ImproperlyConfigured(CONSUL_MISSING)
-
+        #
+        # By default, for correctness, we use a client connection per
+        # operation. If set, self.one_client will be used for all operations.
+        # This provides for the original behaviour to be selected, and is
+        # also convenient for mocking in the unit tests.
+        #
+        self.one_client = None
         self._init_from_params(**parse_url(self.url))
 
     def _init_from_params(self, hostname, port, virtual_host, **params):
         logger.debug('Setting on Consul client to connect to %s:%d',
                      hostname, port)
         self.path = virtual_host
-        self.client = consul.Consul(host=hostname, port=port,
-                                    consistency=self.consistency)
+        self.hostname = hostname
+        self.port = port
+        #
+        # Optionally, allow a single client connection to be used to reduce
+        # the connection load on Consul by adding a "one_client=1" parameter
+        # to the URL.
+        #
+        if params.get('one_client', None):
+            self.one_client = self.client()
+
+    def client(self):
```

Comment:
This could probably just be a `@property`, right?

Reply:
It was originally a property, but that's the bit I had trouble mocking. I tried using both Mock and PropertyMock, but couldn't make the ".kv.get.return_value" work. I tried nested Mock instances, but eventually decided to just use what was right in front of me.

Reply:
Ah, interesting - that's annoying. Thanks for the clarification